controllers/ManagementController.php

```diff
@@ -330,5 +330,67 @@ private function createUserToken()
     {
         return Util::getRandomString(32);
     }
+
+    /**
+     * @param null $id
+     * @throws CannotCreateTokenException
+     * @throws CannotSaveException
+     */
+    public function actionResetPassword($id = null)
+    {
```

Comment:
CSRF : At the beginning of your function, check if the request is POST by throwing a `MethodNotAllowedHttpException` if `\Yii::$app->request->isPost` is `false`